src/mapgen_functions.cpp

```diff
@@ -5381,3 +5390,335 @@ void madd_field( map *m, int x, int y, field_id t, int density )
+                    xn++; coord_rotate_cw( xn, yn, dir ); m->ter_set( xn, yn, t_railroad_tie );
+                    xn++; coord_rotate_cw( xn, yn, dir ); m->ter_set( xn, yn, t_railroad_tie );
+                    xn++; coord_rotate_cw( xn, yn, dir ); m->ter_set( xn, yn, t_railroad_track_on_tie );
+                    xn++; coord_rotate_cw( xn, yn, dir ); m->ter_set( xn, yn, t_railroad_tie );
+                    xn++; coord_rotate_cw( xn, yn, dir ); m->ter_set( xn, yn, t_rubble );
+                    xn++; coord_rotate_cw( xn, yn, dir ); m->ter_set( xn, yn, t_dirt );
+                    xn++; coord_rotate_cw( xn, yn, dir ); m->ter_set( xn, yn, t_dirt );
+                    xn++; coord_rotate_cw( xn, yn, dir ); m->ter_set( xn, yn, t_dirt );
+
+					xn = -1; yn = y + 2;
+
+                    xn++; coord_rotate_cw( xn, yn, dir ); m->ter_set( xn, yn, t_dirt );
+                    xn++; coord_rotate_cw( xn, yn, dir ); m->ter_set( xn, yn, t_dirt );
+                    xn++; coord_rotate_cw( xn, yn, dir ); m->ter_set( xn, yn, t_dirt );
+                    xn++; coord_rotate_cw( xn, yn, dir ); m->ter_set( xn, yn, t_rubble );
+                    xn++; coord_rotate_cw( xn, yn, dir ); m->ter_set( xn, yn, t_rubble );
+                    xn++; coord_rotate_cw( xn, yn, dir ); m->ter_set( xn, yn, t_railroad_track );
+                    xn++; coord_rotate_cw( xn, yn, dir ); m->ter_set( xn, yn, t_rubble );
+                    xn++; coord_rotate_cw( xn, yn, dir ); m->ter_set( xn, yn, t_rubble );
+                    xn++; coord_rotate_cw( xn, yn, dir ); m->ter_set( xn, yn, t_railroad_track );
+                    xn++; coord_rotate_cw( xn, yn, dir ); m->ter_set( xn, yn, t_rubble );
+                    xn++; coord_rotate_cw( xn, yn, dir ); m->ter_set( xn, yn, t_rubble );
+                    xn++; coord_rotate_cw( xn, yn, dir ); m->ter_set( xn, yn, t_dirt );
+                    xn++; coord_rotate_cw( xn, yn, dir ); m->ter_set( xn, yn, t_dirt );
+                    xn++; coord_rotate_cw( xn, yn, dir ); m->ter_set( xn, yn, t_rubble );
+                    xn++; coord_rotate_cw( xn, yn, dir ); m->ter_set( xn, yn, t_rubble );
+                    xn++; coord_rotate_cw( xn, yn, dir ); m->ter_set( xn, yn, t_railroad_track );
+                    xn++; coord_rotate_cw( xn, yn, dir ); m->ter_set( xn, yn, t_rubble );
+                    xn++; coord_rotate_cw( xn, yn, dir ); m->ter_set( xn, yn, t_rubble );
+                    xn++; coord_rotate_cw( xn, yn, dir ); m->ter_set( xn, yn, t_railroad_track );
+                    xn++; coord_rotate_cw( xn, yn, dir ); m->ter_set( xn, yn, t_rubble );
+                    xn++; coord_rotate_cw( xn, yn, dir ); m->ter_set( xn, yn, t_rubble );
+                    xn++; coord_rotate_cw( xn, yn, dir ); m->ter_set( xn, yn, t_dirt );
+                    xn++; coord_rotate_cw( xn, yn, dir ); m->ter_set( xn, yn, t_dirt );
+                    xn++; coord_rotate_cw( xn, yn, dir ); m->ter_set( xn, yn, t_dirt );
```

Comment:
This whole block is terrible and probably copied from somewhere else.
Unless there is a very, very good reason for this, it will need to be all condensed into a neat loop to be tolerable.

And it looks like what you want here is just a separate json.